src/ui/StarRating/StarRating.ts

```diff
@@ -0,0 +1,126 @@
+import 'styling/_StarRating';
+import { $$ } from '../../utils/Dom';
+import { Utils } from '../../utils/Utils';
+import { l } from '../../strings/Strings';
+import { Component } from '../Base/Component';
+import { SVGIcons } from '../../utils/SVGIcons';
+import { exportGlobally } from '../../GlobalExports';
+import { IQueryResult } from '../../rest/QueryResult';
+import { IComponentBindings } from '../Base/ComponentBindings';
+import { ComponentOptions, IFieldOption } from '../Base/ComponentOptions';
+import { Initialization } from '../Base/Initialization';
+
+const DEFAULT_SCALE = 5;
+
+export interface IStarRatingOptions {
+  rating?: IFieldOption;
+  numberOfRatings?: IFieldOption;
+  ratingScale?: number;
+}
+/**
+ * The `StarRating` component renders a 5-star rating widget.
+ *
+ * This component is a result template component (see [Result Templates](https://developers.coveo.com/x/aIGfAQ)).
+ */
+export class StarRating extends Component {
+  static ID = 'StarRating';
+  private rating: number;
+  private numberOfRatings: number;
+  private ratingScale: number;
+
+  static doExport = () => {
+    exportGlobally({
+      StarRating: StarRating
+    });
+  };
+
+  static options: IStarRatingOptions = {
+    /**
+     * Specifies the rating to be displayed as stars.
+     *
+     * Specifying a value for this parameter is required in order for the StarRating component to work.
+     */
+    rating: ComponentOptions.buildFieldOption({ defaultValue: '@rating', required: true }),
+
+    /**
+     * Specifies the number to be displayed in the label indicating the number of ratings.
+     *
+     * If unspecified, no ratings number label is shown. If a value of `0` is provided, a `No Ratings` label is displayed
+     */
+    numberOfRatings: ComponentOptions.buildFieldOption({ required: false }),
```

Comment:
Would renaming to `numberOfRatingsField` be clearer to a user configuring the component?

```
<div class="CoveoStarRating" data-number-of-ratings-field="@some_field"></div>
```

Similar to Olivier, my first intuition was that this option would accept a number.